addons/mail/static/src/messaging/entity/chatter/chatter.js

```diff
@@ -47,13 +55,17 @@ function ChatterFactory({ Entity }) {
             return this.activities.filter(activity => activity.state === 'overdue');
         }
 
-        refresh() {
+        async refresh() {
             const thread = this.thread;
             if (!thread || thread.isTemporary) {
                 return;
             }
             thread.loadNewMessages();
-            thread.fetchAttachments();
+            if (!this._isPreparingAttachmentsLoading && !this.isShowingAttachmentsLoading) {
+                this._prepareAttachmentsLoading();
+            }
+            await thread.fetchAttachments();
+            this._stopAttachmentsLoading();
```

Comment:
Just to point out this is weird that `.fetchAttachments()` doesn't handle prepare/stop attachments loading itself, but I think all this comes from chatter being too smart in current code and most of its logic should be moved to thread.

Reply:
I wondered if it was better for this logic to be in thread or chatter and I went to put it in chatter because the loading is shown in there. I agree that the logic of loaded/not loaded has to be (and already is) in thread but I think it would be weird if the logic of showing/hiding a loader in chatter topbar would be in thread. Don't you think ?

Reply:
the entity "chatter" doesn't really make sense to be honest. we argued its existence for quite some time, and finally decided to have it just for few reasons, which is essentially to make it exist when messaging is not yet ready (and that's necessary because it acts as an interface between views and entities).

All this to say that `chatter` should be minimal and as dumb as possible, and if we do that, then chatter is basically thread (with chatter features generalized to all threads).